logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -783,15 +786,21 @@ internal class ConversationDataSource internal constructor(
             conversationMetaDataDAO.setInformedAboutDegradedMLSVerificationFlag(conversationId.toDao(), isInformed)
         }
 
-    override suspend fun getConversationsWithMembersWithBothDomains(
+    override suspend fun getGroupConversationsWithMembersWithBothDomains(
         firstDomain: String,
         secondDomain: String
-    ): Either<CoreFailure, ConversationsWithMembers> = wrapStorageRequest {
-        val entity = memberDAO.getConversationWithUserIdsWithBothDomains(firstDomain, secondDomain)
-        ConversationsWithMembers(
-            oneOnOne = entity.oneOnOne.mapKeys { it.key.toModel() }.mapValues { it.value.map { userIdEntity -> userIdEntity.toModel() } },
-            group = entity.group.mapKeys { it.key.toModel() }.mapValues { it.value.map { userIdEntity -> userIdEntity.toModel() } }
-        )
+    ): Either<CoreFailure, Map<QualifiedID, List<QualifiedID>>> = wrapStorageRequest {
+        memberDAO.getGroupConversationWithUserIdsWithBothDomains(firstDomain, secondDomain)
+            .mapKeys { it.key.toModel() }
+            .mapValues { it.value.map { userId -> userId.toModel() } }
+    }
+
+    override suspend fun getOneOnOneConversationsWithFederatedMembers(
+        domain: String
+    ): Either<CoreFailure, Map<QualifiedID, QualifiedID>> = wrapStorageRequest {
```

Comment:
What do you think of using here an alias:

`typealias OneToOneMembers =  Map<QualifiedID, QualifiedID>` or something like that ? to be explicit in the Result intent